test/helper.go

```diff
@@ -141,16 +141,16 @@ func NewHelperWithHooksAndDBsetup(t *testing.T, httpServer *httptest.Server, set
 		glog.Fatalf("Unable to initialize testing environment: %s", err.Error())
 	}
 
-	h.CleanDB()
 	h.ResetDB()
 	h.SetupDB(setupDBsqlStatements)
 	env.Start()
 	return h, buildTeardownHelperFn(
-		env.Stop,
 		h.CleanDB,
 		metrics.Reset,
 		stopJWKMockServer,
-		env.Cleanup)
+		env.Cleanup,
+		env.Stop,
```

Comment:
It made more sense to me to "stop" the environment (which asks each `BootService` to _stop_) last.

The code also asks each `BootService` to "clean up" which, IMO, should happen before they stop).

Happy to revert the order of these calls; if others disagree.

Reply:
I've restored the clean-up order.. 

Thinking about it over the weekend `env.Cleanup` should probably be last; after everything has stopped.